hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/AzkabanExecuteFlowTask.groovy

```diff
@@ -0,0 +1,152 @@
+
+    sessionId = AzkabanHelper.resumeOrGetSession(sessionId, azkProject);
+
+    try {
+      //Fetch flows of the project
+      String fetchFlowsResponse = AzkabanClient.fetchProjectFlows(azkProject.azkabanUrl, azkProject.azkabanProjName, sessionId);
+
+      if (fetchFlowsResponse.toLowerCase().contains("error")) {
+        // Check if session has expired. If so, re-login.
+        if (fetchFlowsResponse.toLowerCase().contains("session")) {
+          logger.lifecycle("\nPrevious Azkaban session expired. Please re-login.");
+          executeAzkabanFlow(null);
+        } else {
+          // If response contains other than session error
+          logger.error("Fetching flows from ${azkProject.azkabanUrl} failed. Reason: ${new JSONObject(fetchFlowsResponse).get("error")}");
+        }
+        return;
+      }
+
+      List<String> flows = AzkabanHelper.fetchSortedFlows(new JSONObject(fetchFlowsResponse));
+      if (flows.isEmpty()) {
+        logger.lifecycle("No flows defined in current project");
+        return;
+      }
+
+      List<String> inputFlows;
+
+      if (AzkabanPlugin.interactive) {
+        printFlowsWithIndices(flows);
+        Set<String> indexSet = getFlowIndicesInput();
+        inputFlows = new ArrayList<String>()
+        indexSet.size().times {
+          inputFlows.add(flows.get(Integer.parseInt(indexSet.getAt(it))));
+        }
+      } else {
+        inputFlows = project.getProperties().get("flow").toString().split(",+");
+        for(String flowArg : inputFlows) {
+          if (!flows.contains(flowArg)) {
+            logger.error("Enter correct flow name(s) and try again.");
+            return;
+          }
+        }
+      }
+
+      List<String> responseList = AzkabanClient.batchFlowExecution(azkProject.azkabanUrl, azkProject.azkabanProjName, inputFlows, sessionId);
+      printFlowExecutionResponses(responseList);
+
+    } catch (IndexOutOfBoundsException ex) {
+      logger.error("${ex.getMessage()} : Entered indices not in range, try again.");
+    } catch (Exception ex) {
+      logger.error("Could not Execute Flow.\n${ex.getMessage()}");
+    }
+  }
+
+  /**
+   * Console Input for indices of flows.
+   *
+   * @return indexSet Set of entered indices corresponding to flows
+   */
+  static Set<String> getFlowIndicesInput() {
+    String input = AzkabanHelper.consoleInput(System.console(), " > Enter indices of flows to be executed > ", false);
+    Set<String> indexSet = new HashSet<String>(Arrays.asList(input.split("\\D+")));
+    while(indexSet.isEmpty()) {
+      input = AzkabanHelper.consoleInput(System.console(), " > Enter correct indices of flows to be executed > ", false);
+      indexSet = new HashSet<String>(Arrays.asList(input.split("\\D+")));
+    }
+    return indexSet;
+  }
+
+  /**
+   * Prints the list of flows with Indices.
+   *
+   * @param flows List of flows in the Azkaban Project.
+   */
+  void printFlowsWithIndices(List<String> flows) {
```

Comment:
Hmm... if we move it, it might need to use System.out instead of logger methods, but I think that will still work